backend/open_webui/utils/oauth.py

```
@@ -245,62 +313,89 @@ async def handle_callback(self, provider, request, response):
                         async with aiohttp.ClientSession() as session:
-                            async with session.get(picture_url) as resp:
-                                picture = await resp.read()
-                                base64_encoded_picture = base64.b64encode(
-                                    picture
-                                ).decode("utf-8")
-                                guessed_mime_type = mimetypes.guess_type(picture_url)[0]
-                                if guessed_mime_type is None:
-                                    # assume JPG, browsers are tolerant enough of image formats
-                                    guessed_mime_type = "image/jpeg"
-                                picture_url = f"data:{guessed_mime_type};base64,{base64_encoded_picture}"
+                            async with session.get(picture_url, **get_kwargs) as resp:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)